packages/happy-dom/test/dom-parser/DOMParser.test.ts

```diff
@@ -98,11 +98,19 @@ describe('DOMParser', () => {
 			expect(newDocument.body.textContent).toBe('here is some html elástica ');
 		});
 
-		it('parses SVGs', () => {
+		it('Parses SVGs', () => {
 			const newDocument = domParser.parseFromString(DOMParserSVG, 'image/svg+xml');
 			expect(new XMLSerializer().serializeToString(newDocument).replace(/[\s]/gm, '')).toBe(
 				DOMParserSVG.replace(/[\s]/gm, '')
 			);
 		});
+
```

Comment:
If you're planning on handling all the cases - then i suggest these as well:

```
export const DOMParserBODYSibling = '<body><example></example>Example Text</body><body></body>';
export const DOMParserBODYChildren = '<body><body></body><example></example>Example Text</body>';
		it('recognises does not duplicate body when BODY is a child', () => {
			const newDocument = domParser.parseFromString(DOMParserBODYChildren, 'text/html');
			expect(newDocument.body.innerHTML).toBe('<example></example>Example Text');
		});

		it('recognises does not duplicate body when BODY is a sibling', () => {
			const newDocument = domParser.parseFromString(DOMParserBODYSibling, 'text/html');
			expect(newDocument.body.innerHTML).toBe('<example></example>Example Text');
		});
```

Reply:
I have added the test cases, thank you! :slightly_smiling_face: 

This turned out to be a lot more complicated than what I first thought and I had to refactor quite a bit of the logic, but I have hopefully covered all cases reported and discovered when reading the spec and tested in the browser.

Reply:
It's awesome though that you managed to get this through - impressive work :)

Reply:
Thank you @OlaviSau!